<a href="https://colab.research.google.com/github/amanx98/Flyrank-internship-notebook/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/amanx98/Flyrank-internship-notebook/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### My lane and why

I'm choosing Lane 2: Refresh/Content Opportunity Scoring. My starting question is: given limited review capacity, which pages should be looked at first, and what action might that involve? I'm scoping this initially around the two actions the data gives me real evidence for — refresh and expand — rather than claiming full support across all five (protect/prune/monitor have no baseline criteria defined yet and may need to be explored separately, or dropped, once I look closer). The starter results show a fixed baseline rule reaches only 0.240 precision@50 for flagging pages worth reviewing, while a random forest reaches 0.740 on the same task — evidence there's real signal a simple rule misses.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### The question: decision, action, cost of a wrong call

**Decision:** which pages a content reviewer with limited time looks at first.

**Actor:** a content/SEO reviewer who can only manually check a handful of pages per cycle.

**Action (current scope):** refresh (page shows real signs of decline) or expand (page is thin but pulling real demand). Protect/prune/monitor are possible future actions I don't yet have evidence to support — I may add them later if the data backs them up, rather than assuming they're distinguishable categories from day one.

**Cost of a wrong call:** two kinds. False positive — reviewer spends limited time on a page flagged as urgent that wasn't actually a problem. False negative — a real decliner is missed and keeps losing value until the next cycle. A third risk specific to this lane: acting on a *proxy* decline label that doesn't actually predict a real future outcome, which would waste review capacity on pages that were never really at risk.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd
url = "https://raw.githubusercontent.com/amanx98/Flyrank-internship-notebook/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

In [8]:
# Number 1: how many pages are even in scope (visible + old enough to judge)
in_scope = df[(df["impressions_90d"] > 0) & (df["content_age_days"] >= 90)]
print("Pages in scope:", len(in_scope), "of", len(df))

# Number 2: how many look declining right now (a proxy for "needs attention")
declining = in_scope[in_scope["trend_direction"] == "down"]
print("Declining:", len(declining), f"({len(declining)/len(in_scope):.1%} of in-scope pages)")

# Number 3: how many are thin but still pulling real demand (an "expand" candidate)
thin_but_visible = in_scope[(in_scope["word_count"] < 1200) & (in_scope["impressions_90d"] >= 250)]
print("Thin but visible:", len(thin_but_visible))

Pages in scope: 30000 of 30000
Declining: 16262 (54.2% of in-scope pages)
Thin but visible: 82


### Quick look at the data

**Sample:** 30,000 pages from the starter dataset. All 30,000 passed the in-scope filter (impressions_90d > 0, content_age_days ≥ 90) — this likely means the file was pre-built to only include this population, not that every page in the wider system qualifies. I haven't yet seen what was excluded (newer or zero-impression pages), so I can't say this sample is representative of the full inventory.

| Metric | Count | Share of in-scope |
|---|---:|---:|
| In-scope pages | 30,000 | 100% |
| Flagged declining (`trend_direction == "down"`) | 16,262 | 54.2% |
| Thin but visible (<1200 words, ≥250 impressions) | 82 | 0.3% |

**Why this matters, and what it doesn't prove yet:** refresh candidates outnumber expand candidates by roughly 200:1, which is why I'm scoping to those two actions rather than all five. But 54.2% flagged as "declining" is high enough to be suspicious rather than reassuring — it likely means the label fires on small, short-window wobbles as well as real sustained decline. Before I trust this number for anything beyond scoping, I need to check persistence and magnitude, not just take the flag at face value. Similarly, 82 examples is too thin a base to model statistically — useful as a rule-based flag, but not enough to train anything on without risking pure memorization.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*  


**What this work can say:**
- Which pages *look* like refresh or expand candidates, based on observed signals — decision support, not a verdict.
- A ranked priority order for a reviewer with limited time.
- Directional, observational patterns, not causal claims.

**What this work can never say:**
- That refreshing a page *caused* it to recover — that requires an actual experiment, which this data doesn't give me.
- That any signal here reflects how Google's ranking algorithm actually works.
- That the "declining" label is ground truth — it's a current-window proxy, unvalidated as a real future outcome, and the 54.2% rate suggests it may be noisier than it looks. I plan to check persistence/magnitude before relying on it.
- That protect/prune/monitor are real, distinguishable categories in this data — I haven't found evidence for them yet, and won't claim broad five-action coverage until I do.
- That a high score guarantees a page needs action, or a low score guarantees it doesn't.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.